src/core/src/runtime/itensor.cpp

```diff
@@ -160,7 +160,19 @@ void ITensor::copy_to(const std::shared_ptr<ov::ITensor>& dst) const {
 
     bool finish = false;
     for (size_t dst_idx = 0, src_idx = 0; !finish;) {
-        memcpy(dst_data + dst_idx, src_data + src_idx, src_strides[src_strides.size() - 1]);
+        if (get_element_type() == element::string) {
+            // in case string tensors, it needs to copy of new values for std::string objects
+            // memcpy is not suitable
+            auto dst_string = reinterpret_cast<std::string*>(dst_data + dst_idx);
+            auto src_string = reinterpret_cast<const std::string*>(src_data + src_idx);
+            size_t num_elements_stride = src_strides[src_strides.size() - 1] / element::string.size();
+            for (size_t idx = 0; idx < num_elements_stride; ++idx) {
+                *(dst_string++) = *(src_string++);
+            }
+        } else {
+            memcpy(dst_data + dst_idx, src_data + src_idx, src_strides[src_strides.size() - 1]);
+        }
```

Comment:
let's create a ticket to optimize this function. Currently, it creates quite big `src_strides` vectors which are larger than Tensor itself, secondly it copies more elements than required

Reply:
No, it seems to me this function works as expected. The logic above accumulates dimensions for which there are no strides (or no windows) and compute contiguous chunk of data (with no extra elements) that can be copied by one call.

Reply:
discussed offline, looks normal